src/spikeinterface/sorters/external/kilosort4.py

```diff
@@ -51,14 +56,16 @@ class Kilosort4Sorter(BaseSorter):
         "cluster_downsampling": 20,
         "cluster_pcs": 64,
         "x_centers": None,
-        "duplicate_spike_bins": 7,
+        "duplicate_spike_ms": 0.25,
         "do_correction": True,
         "keep_good_only": False,
         "save_extra_kwargs": False,
         "skip_kilosort_preprocessing": False,
         "scaleproc": None,
         "save_preprocessed_copy": False,
         "torch_device": "auto",
+        "bad_channels": None,
+        "use_binary_file": False,
```

Comment:
Make default None to as "auto"

Reply:
Final change after discussing with Sam: by default `use_binary_file` is `None`, which means that KS will run on the binary file is the recording is binary compatible, otherwise on the wrapper. Setting it to `True` or `False` will force either options, which will require creating a `recording.dat` file in case it's true and the rec is not binary compatible.

For that reason, I added the `delete_recording_dat`, which only deletes a newly created `recording.dat` file (true by default)

Reply:
I think that is a great solution overall!